### PR TITLE
Documentation fixes for the module mapping feature.

### PR DESCRIPTION
**Pull request by vkarak**

### CODE REVIEW

File: docs/running.rst [hunk 1]
```diff
@@ -619,32 +619,76 @@ The asynchronous execution policy may provide significant overall performance be
...
-When invoking ReFrame, user-defined mappings of modules can be passed in order to replace the modules specified in a regression test without the need of modifying the files. This feature enables regression tests to be module-agnostic, which can be very practical in frequent use cases like, for instance, when performing regression tests considering different versions of a module.
+.. versionadded:: 2.11
+
+ReFrame allows you to change the modules loaded by a regression test on-the-fly without having to edit the regression test file.
+This feature is extremely useful when you need to quickly test a newer version of a module, but it also allows you to completely decouple the module names used in your regression tests from the real module names in a system, thus making your test even more portable.
+This is achieved by defining *module mappings*.
+
+There are two ways to pass module mappings to ReFrame.
+The first is to use the ``--map-module`` command-line option, which accepts a module mapping.
+For example, the following line defines maps the module ``test_module`` to the module ``real_module``:
```
Comment (vkarak) [inline docs/running.rst]: I should remove the `defines` here.

Comment (rsarm) [inline docs/running.rst]: 'the following line defines maps the module'.

Maybe 'the following line defines a mapping of the module ... to the module ...'?

File: docs/running.rst [hunk 2]
```diff
@@ -619,32 +619,76 @@ The asynchronous execution policy may provide significant overall performance be
...
+  --map-module 'test_module: real_module0 real_module1'
 
-The other way of passing mappings, is by listing them on a file, which is passed through the command-line option ``--module-mapping``. Each line on the file corresponds to the definition of a mapping for a single module. The format of the mappings on the file is the same as with the option ``--try-module`` and the same rules apply regarding repeated definitions. Text starting with character ``#`` is considered as a comment and empty lines are ignored. The following block shows an example of module mapping file
+The second way of defining mappings is by listing them on a file, which you can then pass to ReFrame through the command-line option ``--module-mappings``.
+Each line on the file corresponds to the definition of a mapping for a single module.
+The syntax of the individual mappings in the file is the same as with the option ``--map-module`` and the same rules apply regarding repeated definitions.
+Text starting with ``#`` is considered a comment and is ignored until the end of line is encountered.
+Empty lines are ignored.
+The following block shows an example of module mapping file
```
Comment (vkarak) [inline docs/running.rst]: I should put a `:` at the end.

File: docs/running.rst [hunk 3]
```diff
@@ -619,32 +619,76 @@ The asynchronous execution policy may provide significant overall performance be
...
+  module-1: module-1a  # an inline comment
   module-2: module-2a module-2b module-2c
-  # module-3: module-3a module-3b
 
-  module-4: module-4a module-4b # module-4c 
+  # This is a full line comment
+  module-4: module-4a module-4b
+
+If both ``--map-module`` and ``--module-mappings`` are passed, ReFrame will first create a mapping from the definitions on the file and it will then process the definitions passed with the ``--map-module`` options.
+As usual later later definitions will override the former.
```
Comment (vkarak) [inline docs/running.rst]: "As usual, later definitions [...]"

[the remark this answers is not in the document]
Comment (vkarak) [reply docs/running.rst]: Yes, I have done the same comment, too.

File: docs/running.rst (hunk 1 above)
Comment (vkarak) [reply docs/running.rst]: I would just remove the `defines`.

**Comment by vkarak** [comment]
@jenkins-cscs retry dom

**Pull request merged**